### PR TITLE
derive ‘append’ and ‘prepend’ from ‘concat’ and ‘of’

### PR DESCRIPTION
**Pull request by davidchambers**
`append` and `prepend` are defined in Sanctuary, but based on <https://github.com/sanctuary-js/sanctuary/pull/414#issuecomment-308724784> we'd like to move the definitions here. :)

### CODE REVIEW

File: test/Maybe.js [hunk 1]
```diff
@@ -27,10 +35,16 @@ Maybe[FL.of] = Maybe.Just;
 
 Maybe[FL.zero] = Maybe[FL.empty];
 
-Maybe.prototype[FL.equals] = function(other) {
+function Maybe$prototype$equals(other) {
```
Comment (gabejohnson) [inline test/Maybe.js]: Why this change?

Comment (davidchambers) [inline test/Maybe.js]: For consistency with changes made in sanctuary-js/sanctuary#388 which were made for the same reason as sanctuary-js/sanctuary#359. Essentially, `Maybe a` satisfies the requirements of Setoid if `a` satisfies the requirements of Setoid.
